WcaOnRails/app/models/competition.rb

```diff
@@ -461,6 +461,14 @@ def unpack_delegate_organizer_ids
     end
   end
 
+  def has_defined_dates?
+    if self.start_date.present? && self.end_date.present?
```

Comment:
I think this can be rewritten as just `return self.start_date.present? && self.end_date.present?`